Remove unnecessary `select` method for `CollectionProxy`

kamipo opened the pull request:
Currently `CollectionProxy` inherits `Relation` and `Relation` includes
`QueryMethods`. This method is completely duplicated.

https://github.com/rails/rails/blob/v5.0.0/activerecord/lib/active_record/relation/query_methods.rb#L271-L275

rafaelfranca commented on activerecord/lib/active_record/associations/collection_proxy.rb:
```diff
@@ -54,63 +54,6 @@ def loaded?
         @association.loaded?
       end
 
-      # Works in two ways.
```
We should keep the documentation even that we are deleting the implementation. I think using `cal-seq directive we can keep the documentation and remove the implementation.

kamipo committed:
Remove unnecessary `select` method for `CollectionProxy`

Currently `CollectionProxy` inherits `Relation` and `Relation` includes
`QueryMethods`. This method is completely duplicated.

https://github.com/rails/rails/blob/v5.0.0/activerecord/lib/active_record/relation/query_methods.rb#L271-L275

kamipo commented on activerecord/lib/active_record/associations/collection_proxy.rb:
```diff
@@ -53,6 +53,12 @@ def loaded?
+      # :method: select
+      #
+      # :call-seq:
+      #   select(*fields, &block)
```
I changed to keep the doc.
